src/components/Table.js

```diff
@@ -241,46 +241,72 @@ export default class Table extends React.Component<TableDefaultProps, TableProps
   }
 
   componentWillReceiveProps(newProps: TableProps) {
-    const { rows, noEmptySelection } = newProps;
+    const { rows, noEmptySelection, rowComparator, onSelect } = newProps;
     const { rows: prevRows } = this.props;
+    const { selectedIndices: prevSelectedIndices, activeRowIndex } = this.state;
 
     // If a rowComparator is specified, use it, otherwise, do a deep row comparison and reset selection if anything changed.
     const shouldUpdateRows = !isEqual(rows, prevRows);
-    const shouldResetSelection = rows.length !== prevRows.length
-      || (newProps.rowComparator && !isEqualWith(rows, prevRows, newProps.rowComparator))
-      || (!newProps.rowComparator && shouldUpdateRows);
+    let shouldResetSelection = (rowComparator && !isEqualWith(rows, prevRows, rowComparator))
+      || (!rowComparator && shouldUpdateRows);
 
     if (shouldResetSelection || shouldUpdateRows) {
+      const updatedSelectedRows = [];
+
       const sortedRows = rows && rows.length > 0
         ? rows.map((row, tableRowIndex) => {
-          return { ...row, tableRowIndex };
+          const updatedRow = { ...row, tableRowIndex };
+          if (prevSelectedIndices.has(tableRowIndex)) {
+            updatedSelectedRows.push(updatedRow);
+          }
+          return updatedRow;
         })
         : [];
-      const selectedIndices = sortedRows.length > 0 && noEmptySelection
+
+      const numPreviousSelections = prevSelectedIndices.size;
+      const numUpdatedSelections = updatedSelectedRows.length;
+
+      // Indicates one of the previously selected rows can no longer be found.
+      if (numPreviousSelections !== numUpdatedSelections) {
+        shouldResetSelection = true;
+      }
+
+      const resetIndices = sortedRows.length > 0 && noEmptySelection
         ? new Set([0])
         : new Set([]);
-      const anchorRowIndex = sortedRows.length > 0 && noEmptySelection
+      const resetIndex = sortedRows.length > 0 && noEmptySelection
```

Comment:
If a row that had been selected is no longer in the table, but the previous anchor row is still there, won't this change the anchor row? I could be missing something—there are a lot of moving parts... :)

Reply:
If the previously selected row is no longer in the table, we _want_ to reset the anchor row. The anchor row is only meaningful in relation to the selected row.

Reply:
Oh, I think I was thinking that the anchor row was the one shown in the inspector... OK.

Reply:
Ah, gotchya. The `activeRowIndex` indicates the row that's displayed in the inspector. The `anchorRowIndex` is an internal tracking mechanism for multiselect keyboard actions to determine direction and boundaries.